ivy/utils/backend/ast_helpers.py

```diff
@@ -174,24 +174,44 @@ def visit_ImportFrom(self, node):
         else:
             return _parse_relative_fromimport(node)
 
-    def impersonate_import(self, tree: ast.Module):
+    def impersonate_import(self, tree: ast.Module, local_ivy_id=None):
         if self.include_ivy_import:
-            tree.body.insert(
-                self.insert_index,
-                ast.ImportFrom(
-                    module=importlib_module_path,
-                    names=[ast.alias(name=importlib_from_import_fn, asname=None)],
-                    level=0,
-                ),
-            )
-            tree.body.insert(
-                self.insert_index,
-                ast.ImportFrom(
-                    module=importlib_module_path,
-                    names=[ast.alias(name=importlib_abs_import_fn, asname=None)],
-                    level=0,
-                ),
-            )
+            if local_ivy_id is not None:
```

Comment:
I think the following logic can be simplified/unified to look a bit cleaner (the `ast.parse` is such a nice move tbh makes the code so much more readable), anyway, so we want to have this overall effect:
`if local_ivy_id is not None:`
we want this to be slapped at the beginning of the file
```python
import ivy
_absolute_import = ivy.utils.backend.handler._compiled_backends_ids[{local_ivy_id}].utils._importlib._absolute_import
_from_import = ivy.utils.backend.handler._compiled_backends_ids[{local_ivy_id}].utils._importlib._from_import
```

`else:`
we slap this
```python
from ivy.utils.backend.handler.utils._importlib import _absolute_import
from ivy.utils.backend.handler.utils._importlib import _from_import
```
we can merge both cases into one
import ivy

`if local_ivy_id is not None`:
```python
import ivy
_absolute_import = ivy.utils.backend.handler._compiled_backends_ids[{local_ivy_id}].utils._importlib._absolute_import
_from_import = ivy.utils.backend.handler._compiled_backends_ids[{local_ivy_id}].utils._importlib._from_import
```

`else`:
```python
import ivy
_absolute_import = ivy.utils.backend.handler.utils._importlib._absolute_import
_from_import = ivy.utils.backend.handler.utils._importlib._from_import
```

code-wise it would look like this:
```python
# ast.parse produces a tree with Module tree, using body[0]
# we can obtain the specific node that we require.
ivy_import_node = ast.parse("import ivy").body[0]
import_stmt = "ivy.utils.backend.handler"
if local_ivy_id is not None:
    import_stmt += f"._compiled_backends_ids[{local_ivy_id}]"
abs_import_stmt = import_stmt + ".utils._importlib._absolute_import"
from_import_stmt = import_stmt + ".utils._importlib._from_import"


abs_import_node = ast.parse(abs_import_stmt).body[0]
from_import_node = ast.parse(from_import_stmt).body[0]
tree.body.insert(self.insert_index, from_import_node)
tree.body.insert(self.insert_index, abs_import_node)
tree.body.insert(self.insert_index, ivy_import_node)

return tree
```

Reply:
This code is definitely shorter, but makes it harder for whoever reads the code to actually know what happens right away, I don't think the tradeoff is worth it, also when debugging, `from x import y` may be more intuitive than `import x; y = x.y`

Reply:
I actually think it is more readable, your point about `from x import y` being more intuitive is 100% correct, but in this case, we aren't writing code that is explicitly shown, but implicitly injected to do a specific purpose (it is not visible) so I think it is totally fine, a comment (or comments) can be added to explain what is going regardless (whether we go with this way or not actually, a few comments will be needed), I really think this is more concise and easier to debug.